File: android/app/src/main/java/net/bradball/teetimecaddie/android/ui/common/avatars/TtcAvatarPlaceholder.kt
package net.bradball.teetimecaddie.android.ui.common.avatars

import net.bradball.teetimecaddie.android.theme.TtcColorRole

/**
 * A non-identity [TtcAvatar] affordance: [Guest] (a dashed circle with a person glyph) or [Empty]
 * (a dashed circle with an add glyph). These ignore [TtcColorRole] — they render the design's dashed
 * `outlineVariant` placeholder treatment.
 *
 * Distinct from a [TtcColorRole.Neutral] avatar, which still shows initials: use a placeholder when
 * there is no player to name at all.
 *
 * The iOS twin is `TtcAvatarPlaceholder` in `ui/common/Avatars/TtcAvatarPlaceholder.swift`.
 */
enum class TtcAvatarPlaceholder { Guest, Empty }

File: ios/TeeTimeCaddie/TeeTimeCaddie/Source/ui/common/Avatars/TtcAvatarPlaceholder.swift
import Foundation

/// A non-identity ``TtcAvatar`` affordance: `.guest` (a dashed circle with a person glyph) or
/// `.empty` (a dashed circle with an add glyph). These ignore ``TtcColorRole`` — they render the
/// design's dashed `outlineVariant` placeholder treatment.
///
/// Distinct from a `.neutral` avatar, which still shows initials: use a placeholder when there is no
/// player to name at all.
///
/// The Android twin is `TtcAvatarPlaceholder` in `ui/common/avatars/TtcAvatarPlaceholder.kt`.
enum TtcAvatarPlaceholder {
    case guest
    case empty
}
